src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -336,164 +350,195 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
         return uuidUtils.uuidFromBase64(transactionId)
                 .fold(
                         Mono::error,
-                        transactionIdDecoded -> Mono.just(transactionIdDecoded)
-                                .doOnNext(el -> log.info("decoded transaction id: {}", transactionIdDecoded))
-                                .flatMap(
-                                        el -> transactionsViewRepository
-                                                .findById(transactionIdDecoded.toString())
-                                                .switchIfEmpty(
-                                                        Mono.error(
-                                                                new TransactionNotFoundException(
-                                                                        transactionIdDecoded.toString()
+                        transactionIdDecoded -> {
+                            TransactionId trnId = new TransactionId(transactionIdDecoded);
+                            log.info("decoded transaction id: {}", transactionIdDecoded);
+                            Mono<BaseTransaction> baseTransaction = transactionsUtils.reduceEvents(trnId);
+                            return wasTransactionAuthorized(trnId)
+                                    .flatMap(alreadyAuthorized -> {
+                                        if (Boolean.FALSE.equals(alreadyAuthorized)) {
+                                            return Mono.just(
+                                                    Either.<TransactionInfoDto, Mono<BaseTransaction>>right(
+                                                            baseTransaction
+                                                    )
+                                            );
+                                        } else {
+                                            return baseTransaction.flatMap(
+                                                    trx -> {
+                                                        log.info(
+                                                                "Transaction authorization outcome already received. Transaction status: {}",
+                                                                trx.getStatus()
+                                                        );
+                                                        return Mono.just(
```

Comment:
In both branches you're using `Mono.just` inside a `flatMap` => can be simplified to `map`